src/shared/common/forms/__tests__/dropdown.spec.js

```diff
@@ -64,5 +86,36 @@ describe('Form dropdown component', () => {
         expect(mockCallback).toHaveBeenCalledTimes(2);
         expect(mockCallback).toHaveBeenCalledWith({ 'dropdown': secondValue });
     });
-});
 
+    test('should render only active options (exclie inactive dropdown value), default is active', () => {
+        const wrapperConditionA = render(<Dropdown name="radio-group" conditionChoices={conditionChoices}
+            data={{ 'conditionValue': 'a' }} updateState={() => null} />);
+
+        expect(wrapperConditionA).toBeDefined();
+
+        expect(screen.getByText('conditionA valueA')).toBeInTheDocument();
+        expect(screen.queryByText('conditionA valueB')).toBeInTheDocument();
+        expect(screen.queryByText('conditionA valueC')).not.toBeInTheDocument();
+
+        const wrapperConditionB = render(<Dropdown name="radio-group" conditionChoices={conditionChoices}
+            data={{ 'conditionValue': 'b' }} updateState={() => null} />);
+
+        expect(wrapperConditionB).toBeDefined();
+
+        expect(screen.getByText('conditionB valueA')).toBeInTheDocument();
+        expect(screen.getByText('conditionB valueB')).toBeInTheDocument();
+        expect(screen.queryByText('conditionB valueC')).not.toBeInTheDocument();
```

Comment:
We need to be testing that condition B values are not shown when condition A is selected.
```suggestion
        expect(screen.getByText('conditionB valueA')).not.toBeInTheDocument();
        expect(screen.getByText('conditionB valueB')).not.toBeInTheDocument();
        expect(screen.queryByText('conditionB valueC')).not.toBeInTheDocument();
```

I agree it's a good idea to check that the options change when the value changes. React testing library provides a way to rerender an existing component. https://testing-library.com/docs/example-update-props/

So in this case you'd need to add something like: 
```js
       // ... at end of exitsing test method, rename wrapperConditionA back to just wrapper
       wrapper.rerender(<Dropdown name="radio-group" conditionChoices={conditionChoices}
            data={{ 'conditionValue': 'b' }} updateState={() => null} />);

       expect(screen.getByText('conditionA valueA')).not.toBeInTheDocument();
       expect(screen.getByText('conditionB valueA')).toBeInTheDocument();
```